fluent/src/resolver.js

```diff
@@ -289,11 +289,12 @@ export function resolveComplexPattern(scope, ptn) {
           `(${part.length}, max allowed is ${MAX_PLACEABLE_LENGTH})`
         )
       );
-      result.push(part.slice(MAX_PLACEABLE_LENGTH));
-    } else {
-      result.push(part);
+      scope.dirty.delete(ptn);
+      return new FluentNone();
```

Comment:
This isn't perfect, but does provide some level of protection. The problem this approach still suffers from is that `FluentNone` will then `toString`-ed to `{???}` many many times over, depending on how many patterns fail in references up the nesting tree. This also means that the number of accumulated errors may be huge.

I filed https://github.com/projectfluent/fluent/issues/277 to continue the discussion in the spec.